### DISCUSSION TITLE
Ability to group fields in the output

### DISCUSSION THREAD

**Discussion by karambaq** [pydantic/pydantic#8233] [Question]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

I have model which looks like this:

```python
class Game(Base):

    id = Column(Integer, primary_key=True)

    a = Column(String(50))
    b = Column(String(50))
```

And the schema:

```python
class GameBase(BaseModel):
    id: int
    a: str
    b: str
```

And resulting json is:

```json
{
    "id": ...,
    "a": ...,
    "b": ...,
}
```

But I need it to look like this:

```json
{
    "id": ...,
    "group": {
        "a": ...,
        "b": ...,
    }
}
```
---

How can I achieve this?

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [X] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [X] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [X] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
Hi @karambaq,

Thanks for your question. I'm going to move this to a discussion, as it's more of a question than a feature request.

I'd recommend you check out the documentation on customizing JSON schema, which you can find here: https://docs.pydantic.dev/dev/concepts/json_schema/#modifying-the-schema. We should have a PR across the line later today that improves these docs even more 😄.

I'd recommend you customize `__get_pydantic_json_schema__` or customize the JSON schema generation process by modifying the `GenerateJsonSchema` class.
